src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -153,10 +153,6 @@ private static PackageSpec GetPackageSpec(JsonTextReader jsonReader, string name
                         ReadMSBuildMetadata(jsonReader, packageSpec);
                         break;
 
-                    case "restoreSettings":
```

Comment:
Why restoreSettings are not needed anymore?

Reply:
There's no writer for these. 

The writer was removed in: 7b9f4c1#diff-f585ed8fd184026effea6796bc44f32f - Jan 19, 2018

.